pkg/storage/helpers_test.go

```diff
@@ -232,6 +233,28 @@ func (r *Replica) DescLocked() *roachpb.RangeDescriptor {
 	return r.mu.state.Desc
 }
 
+// GetGCThreshold returns the range's GCThreshold, acquiring a replica lock in
+// the process.
+func (r *Replica) GetGCThreshold() hlc.Timestamp {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return r.mu.state.GCThreshold
+}
+
+// GetTxnSpanGCThreshold returns the range's TxnSpanGCThreshold, acquiring a replica lock in
+// the process.
+func (r *Replica) GetTxnSpanGCThreshold() hlc.Timestamp {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return r.mu.state.TxnSpanGCThreshold
+}
+
+func (r *Replica) AssertState(ctx context.Context, reader engine.Reader) {
+	r.mu.RLock()
```

Comment:
You should acquire `raftMu` (before `r.mu`) as well. Otherwise you can read inconsistent state where on-disk has already been updated, but in-memory has not.